Deprecating old technology (#1845)
elastic/elasticsearch-hadoop: docs/src/reference/asciidoc/appendix/breaking.adoc

ES-hadoop in a low memory environment, consider lowering `es.scroll.size` to
avoid issues.

> ==== Deprecations
> 
> The following are deprecated and will be removed in a future release:

Reviewer: I'd steal some of the wording from the [Elasticsearch migration guide here](https://www.elastic.co/guide/en/elasticsearch/reference/8.0/migrating-8.0.html#deprecated-8.0).

I'd also bump the heading up so that it's at the same level as the 8.0 breaking changes. That made the on-page TOC a little more readable to me.

This also adjusts the tags so they only include the changes/deprecations.

```suggestion
// end::notable-v8-breaking-changes[]

[[deprecations-8.0]]
=== Deprecations in 8.0

The following functionality has been deprecated in {eh} 8.0 and will be removed
in a future version. While this won’t have an immediate impact on your
applications, we strongly encourage you take the described steps to update your
code after upgrading to 8.0.

// tag::notable-v8-breaking-changes[]
```

> ===== Spark 1.x

Reviewer: Rather than rely on the previous heading, I'd add "is deprecated" so this is a little more scannable.
```suggestion
==== Spark 1.x support is deprecated
```

Spark 1.x is no longer maintained. Spark 2 and Spark 3 are still supported.

> ===== Scala 2.10 support for Spark 2.x

Reviewer: ```suggestion
==== Scala 2.10 support for Spark 2.x is deprecated
```

Spark deprecated support for Scala 2.10 in the Spark 2.0.0 release and removed it in the 2.3.0 release. Scala 2.11 and 2.12 are supported
for Spark 2.x.

> ===== Hadoop 1

Reviewer: ```suggestion
==== Hadoop 1.x support is deprecated
```

Hadoop 1.x is no longer maintained. Support for it has not been tested since {eh} 6.0. Support is now formally deprecated. Hadoop 2 and
Hadoop 3 are supported.

> ===== Pig

Reviewer: "Pig" and "Storm" looked funny to me without a preceding "Apache" (at least on first use).
```suggestion
==== Apache Pig support is deprecated
```

Apache Pig is no longer maintained.

> ===== Storm

Reviewer: ```suggestion
==== Apache Storm support is deprecated
```
